src/deepsparse/transformers/pipelines.py

```diff
@@ -0,0 +1,992 @@
+            model, it will be split in several chunks with some overlap. This argument
+            controls the size of that overlap
+        :param max_answer_len: maximum length of predicted answers (e.g., only
+            answers with a shorter length are considered)
+        :param max_seq_len: maximum length of the total sentence (context + question)
+            after tokenization. The context will be split in several chunks
+            (using the doc_stride) if needed
+        :param max_question_len: maximum length of the question after tokenization.
+            It will be truncated if needed
+        :param handle_impossible_answer: whether or not we accept impossible as an
+            answer
+        :param num_spans: maximum number of span to use as input from a long
+            context. Default is to stride the entire context string
+        :param preprocessed_inputs: if provided, preprocessing will be skipped in favor
+            of these inputs. Expected format is the output of self.preprocess; a tuple
+            of (examples, features_list)
+        :return: dict or list of dictionaries, each containing the following keys:
+            `"score"` - The probability associated to the answer
+            `"start"` - The start index of the answer
+            `"end"` - The end index of the answer
+            `"answer"` - The answer to the question
+        """
+        # Set defaults values
+        kwargs.setdefault("topk", 1)
+        kwargs.setdefault("max_answer_len", 15)
+        kwargs.setdefault("handle_impossible_answer", False)
+        kwargs.setdefault("preprocessed_inputs", None)  # (examples, features_list)
+
+        if kwargs["topk"] < 1:
+            raise ValueError(f"topk parameter should be >= 1 (got {kwargs['topk']})")
+
+        if kwargs["max_answer_len"] < 1:
+            raise ValueError(
+                "max_answer_len parameter should be >= 1 "
+                f"(got {kwargs['max_answer_len']})"
+            )
+
+        # run pre-processing if not provided
+        examples, features_list = kwargs["preprocessed_inputs"] or self.preprocess(
+            *args, **kwargs
+        )
+
+        # forward pass and post-processing
+        all_answers = []
+        for features, example in zip(features_list, examples):
+            model_input_names = self.tokenizer.model_input_names + ["input_ids"]
+            fw_args = {
+                k: [feature.__dict__[k] for feature in features]
+                for k in model_input_names
+            }
+
+            # Manage tensor allocation on correct device
+            fw_args = {k: np.array(v) for (k, v) in fw_args.items()}
+            start, end = self._forward(fw_args)[:2]
+
+            # TODO: torch
+            # fw_args = {k: torch.tensor(v, device=self.device)
+            #   for (k, v) in fw_args.items()}
+            # start, end = self.model(**fw_args)[:2]
+            # start, end = start.cpu().numpy(), end.cpu().numpy()
```

Comment:
nit: remove TODOs
```suggestion
```

Reply:
going to keep this in for now in case torch, is added in the near future. Will delete if taken off the roadmap